src/rmkit/fb/fb.cpy

```diff
@@ -256,19 +265,30 @@ namespace framebuffer:
       if o_y >= self.height || o_x >= self.width || o_y < 0 || o_x < 0:
         return
 
-      update_dirty(dirty_area, o_x, o_y)
-      update_dirty(dirty_area, o_x+w, o_y+h)
+      if likely(fill):
```

Comment:
It looks like almost every function calls this with `fill=true`. I think the only place where `fill=false` would happen is in the original `draw_rect` itself right?

What about making `draw_rect` include something like this? (I haven't tested, just a thought)

```cpp
if fill:
  _draw_rect_fast(x, y, w, h, ...)
else:
  _draw_rect_fast(x, y, w, 1, ...)     // top
  _draw_rect_fast(x, y+h-1, w, 1, ...) // bottom
  _draw_rect_fast(x, y, 1, h, ...)     // left
  _draw_rect_fast(x+w-1, y, 1, h, ...) // right
```

The main difference I think is that do_dithering could still be inlined in `_draw_rect_fast` since it would only have one call site.

Reply:
that looks good to me